### ISSUE TITLE
Undead Slayer is applying to non-undead targets if the attacker with the feat has the undead trait

### ISSUE THREAD

**Issue by unindel** [foundryvtt/pf2e#10250]
My group is playing a game of Blood Lords and has a Skeleton Monk with Dhampir heritage that has the Undead Slayer level 5 ancestry feat.

This feat is supposed to add a circumstance bonus to attacks against undead creatures (doubled against vampires). The feat works fine when I click the buttons from the character sheet for damage (when an undead is targeted it adds +2 since he has two weapon dice, and +4 against a vampire targeted) but if I use the "Damage" or "Critical" buttons from the chat attack roll, it seems to apply the undead slayer to even non-undead targets. If I remove the "Undead" trait from the player's Skeleton ancestry, the feat works as intended from either the character sheet or the chat roll so we're using that as a workaround for now.

So to summarize:
1. PC with Undead Slayer ancestry feat + skeleton ancestry
2. Target a living enemy, click attack on fist attack.
3. From the chat result, select the damage roll button. Note the damage roll has added the Undead Slayer bonus.
4. Roll damage for the fist attack from the character sheet. Note the damage roll does not have the Undead Slayer bonus
5. Remove the Undead trait from the PC's skeleton ancestry item.
6. Retry step 3 and 4, note that this time it works in both cases, triggering just off the target's undead/vampire trait.

**Comment by stwlam** [comment]
Foundry/system versions?

**Comment by unindel** [comment]
It was foundry 11.308 and system 5.5.2 but I will update tomorrow and try again.
